libs/diag/diagnostic.go

```diff
@@ -96,7 +99,14 @@ func (ds Diagnostics) HasError() bool {
 func (ds Diagnostics) Error() error {
 	for _, d := range ds {
 		if d.Severity == Error {
-			return fmt.Errorf(d.Summary)
+			message := d.Detail
+			if message == "" {
+				message = d.Summary
+			}
+			if d.ID != "" {
+				message = fmt.Sprintf("%s: %s", d.ID, message)
```

Comment:
This should use a string writer or simple concatenation. The message may contain % itself.